File: FrontEnd/src/app/features/billing/invoices/classes/dtos/invoice-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { AadeVM } from '../view-models/aade-vm'
import { CustomerVM } from '../view-models/customer-vm'
import { DocumentTypeVM } from '../view-models/documentType-vm'
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'
import { PortReadDto } from './port-read-dto'

export interface InvoiceReadDto extends Metadata {

    // PK
    invoiceId: Guid
    // Fields
    date: string
    no: number
    // Object fields
    customer: CustomerVM
    destination: SimpleEntity
    documentType: DocumentTypeVM
    paymentMethod: SimpleEntity
    ship: SimpleEntity
    aade: AadeVM
    ports: PortReadDto[]
    // Fields
    remarks: string
    // Persons
    adults: number
    kids: number
    free: number
    totalPax: number
    // Amounts
    netAmount: number
    vatPercent: number
    vatAmount: number
    grossAmount: number

}

File: API/Features/Billing/Invoices/Dtos/InvoiceReadDto.cs
using System;
using System.Collections.Generic;
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Billing.Invoices {

    public class InvoiceReadDto : IMetadata {

        // PK
        public Guid InvoiceId { get; set; }
        // Fields
        public DateTime Date { get; set; }
        public int No { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPax { get; set; }
        public decimal NetAmount { get; set; }
        public decimal VatPercent { get; set; }
        public decimal VatAmount { get; set; }
        public decimal GrossAmount { get; set; }
        public string Remarks { get; set; }
        // Child tables
        public InvoiceReadAadeDto Aade { get; set; }
        public List<InvoicePortReadDto> Ports { get; set; }
        // Navigation
        public InvoiceCustomerReadDto Customer { get; set; }
        public SimpleEntity Destination { get; set; }
        public InvoiceDocumentTypeReadDto DocumentType { get; set; }
        public SimpleEntity PaymentMethod { get; set; }
        public SimpleEntity Ship { get; set; }
        public SimpleEntity Port { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}